apiserver/pkg/codeflare/appwrapper_builder.go

```diff
@@ -0,0 +1,167 @@
+	RayClusterFromAppWrapper(clusterAppWrapper *mcadApi.AppWrapper) (*rayv1api.RayCluster, error)
+	AppWrapperFromRayJob(rayJob *rayv1api.RayJob) (*mcadApi.AppWrapper, error)
+	RayJobFromAppWrapper(jobAppWrapper *mcadApi.AppWrapper) (*rayv1api.RayJob, error)
+}
+
+type defaultAppwrapperConverter struct {
+	AppWrapperConverter
+	scheme *runtime.Scheme
+}
+
+func NewAppWrapperConverter(scheme *runtime.Scheme) AppWrapperConverter {
+	return &defaultAppwrapperConverter{scheme: scheme}
+}
+
+func (awb *defaultAppwrapperConverter) AppWrapperFromRayCluster(rayCluster *rayv1api.RayCluster) (*mcadApi.AppWrapper, error) {
+	output, err := awb.rayClusterToYAML(rayCluster)
+	if err != nil {
+		return nil, fmt.Errorf("unable to serialize cluster to yaml: %w", err)
+	}
+	requests, limits := awb.calculateClusterAggregateResources(&rayCluster.Spec.HeadGroupSpec, rayCluster.Spec.WorkerGroupSpecs)
+	clusterAppWrapper := awb.makeAppWrapper(rayCluster.Name, rayCluster.Namespace, requests, limits, output)
+	return clusterAppWrapper, nil
+}
+
+func (awb *defaultAppwrapperConverter) RayClusterFromAppWrapper(clusterAppWrapper *mcadApi.AppWrapper) (*rayv1api.RayCluster, error) {
+	bytez := clusterAppWrapper.Spec.AggrResources.GenericItems[0].GenericTemplate.Raw
+	codec := serializer.NewCodecFactory(awb.scheme, serializer.EnablePretty).LegacyCodec(rayv1api.SchemeGroupVersion)
+	rayCluster := &rayv1api.RayCluster{}
+	if err := runtime.DecodeInto(codec, bytez, rayCluster); err != nil {
+		return nil, err
+	}
+	return rayCluster, nil
+}
+
+func (awb *defaultAppwrapperConverter) AppWrapperFromRayJob(rayJob *rayv1api.RayJob) (*mcadApi.AppWrapper, error) {
+	output, err := awb.rayJobToYAML(rayJob)
+	if err != nil {
+		return nil, fmt.Errorf("unable to serialize cluster to yaml: %w", err)
+	}
+	requests, limits := awb.calculateJobAggregateResources(rayJob)
+	jobAppWrapper := awb.makeAppWrapper(rayJob.Name, rayJob.Namespace, requests, limits, output)
+	return jobAppWrapper, nil
+}
+
+func (awb *defaultAppwrapperConverter) RayJobFromAppWrapper(jobAppWrapper *mcadApi.AppWrapper) (*rayv1api.RayJob, error) {
+	bytez := jobAppWrapper.Spec.AggrResources.GenericItems[0].GenericTemplate.Raw
+	codec := serializer.NewCodecFactory(awb.scheme, serializer.EnablePretty).LegacyCodec(rayv1api.SchemeGroupVersion)
+	rayJob := &rayv1api.RayJob{}
+	if err := runtime.DecodeInto(codec, bytez, rayJob); err != nil {
+		return nil, err
+	}
+	return rayJob, nil
+}
+
+func (awb *defaultAppwrapperConverter) calculateClusterAggregateResources(clusterHeadSpec *rayv1api.HeadGroupSpec, workerSpecs []rayv1api.WorkerGroupSpec) (corev1.ResourceList, corev1.ResourceList) {
+	var replicaCount int
+	requests := corev1.ResourceList{}
+	limits := corev1.ResourceList{}
+	awb.addContainersResources(clusterHeadSpec.Template.Spec.Containers, requests, limits)
+	awb.addContainersResources(clusterHeadSpec.Template.Spec.InitContainers, requests, limits)
+
```

Comment:
We do not support init containers in the cluster definition, they are inserted later by the operator, besides this is minimal
What I suggest here is to drop this, but add autoscaler, if it is defined